internal/venice-common/src/main/java/com/linkedin/venice/controller/converters/RequestConverter.java

```diff
@@ -0,0 +1,15 @@
+package com.linkedin.venice.controller.converters;
+
+import com.linkedin.venice.controller.requests.ControllerRequest;
+
+
+/**
+ * A general interface to represent request converters that operates on {@Link T} to produce a transport specific
+ * request {@link D}
+ * @param <T> Source controller request type
+ * @param <D> Transport specific request type
```

Comment:
What do `T` and `D` stand for? Although it is fine to use single letters for the generic type names, I think it's important to try to make these names as self-explanatory as possible. For example, `I`/`O` which stand for Input/Output might be fine. It is also fine to spell out whole words instead of using single letters, e.g. `SOURCE`/`DESTINATION`, or `SRC`/`DEST`.

Reply:
T -> The controller request type (e.g., `storeCreateRequest`, `jobStatusRequest`)
D -> controller transport specific request type (e.g., http, grpc etc)

I did use some of elaborative naming for generic types in few places in the code where I saw no ambiguity. I thought a bit and then wasn't fully convinced if `SRC` and `DEST` was going convey its intent; I even wondered if `SRCREQ` to `DESTREQ` would help.

Given this converter interface is not a general converter interface rather specific to the controller requests, I found the above choices too generic and hence resorted write the parameter java document to provide some clarity.  Do we have any convention on naming generic types within our code base?

For e.g., Can we have something like `ControllerRequestType` `TransportRequestType`?

Reply:
`T` usually stands for "Type", which is fine if there is just a single thing to maintain a generic "type" for. In this case there are too things, however, so `T` seems ambiguous. `D` might stand for "Destination" in this context, though I'm not sure?

Reply:
ah i see; `T` is your concern as it by default has the type connotation. Let me update `ControllerRequestType` and `TransportRequestType` if you have no objections. If not, I can fallback to `SRC` and `DEST` which i think is okay along with the java docs